src/liballoc/vec.rs

```diff
@@ -358,6 +358,30 @@ impl<T> Vec<T> {
         }
     }
 
+    /// Decomposes a `Vec<T>` into its raw components.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(vec_into_raw_parts)]
+    /// let v = vec![1, 2, 3];
+    ///
+    /// let (ptr, len, cap) = v.into_raw_parts();
+    ///
+    /// let rebuilt = unsafe { Vec::from_raw_parts(ptr, len, cap) };
+    /// assert_eq!(rebuilt, [1, 2, 3]);
+    /// ```
```

Comment:
Might be a good idea to do a "transmute" via raw ptrs here? After all that was your original motivation.

Also please add a FIXME at the `transmute` docs to use `into_raw_parts` once that's stable.

Reply:
Seems reasonable, but I don't want to get too bogged down in the transmute aspects — do you have any ideas for a "cognitively lightweight" transmute? I'm thinking just `i32` -> `u32`.

Reply:
Yeah, switching the signedness sounds like a good candidate.